usr/sbin/p11sak/p11sak.c

```diff
@@ -14,3457 +14,1235 @@
+                .value.plain = &opt_long, },
+      .description = "List keys in long (detailed) format.", },
+    { .short_opt = 0, .long_opt = "detailed-uri", .required = false,
+      .long_opt_val = OPT_DETAILED_URI,
+      .arg =  { .type = ARG_TYPE_PLAIN, .required = false,
+                .value.plain = &opt_detailed_uri, },
+      .description = "Show detailed PKCS#11 URI.", },
+    { .short_opt = 0, .long_opt = NULL, },
+};
+
+#define null_generic_args           NULL
+#define null_aes_args               NULL
+#define null_aes_xts_args           NULL
+#define null_rsa_args               NULL
+#define null_dh_args                NULL
+#define null_dsa_args               NULL
+#define null_ec_args                NULL
+#define null_ibm_dilithium_args     NULL
+#define null_ibm_kyber_args         NULL
+
+static const struct p11sak_enum_value p11sak_list_remove_key_keytypes[] = {
+    KEYGEN_KEYTYPES(null),
+    GROUP_KEYTYPES,
+    { .value = NULL, },
+};
+
+static const struct p11sak_arg p11sak_list_key_args[] = {
+    { .name = "KEYTYPE", .type = ARG_TYPE_ENUM, .required = false,
+      .enum_values = p11sak_list_remove_key_keytypes,
+      .value.enum_value = &opt_keytype,
+      .description = "The type of the keys to list (optional). If no key type "
+                     "is specified, all key types are listed.", },
+    { .name = NULL },
+};
+
+static const struct p11sak_opt p11sak_remove_key_opts[] = {
+    PKCS11_OPTS,
+    FILTER_OPTS,
+    { .short_opt = 'f', .long_opt = "force", .required = false,
+      .arg =  { .type = ARG_TYPE_PLAIN, .required = false,
+                .value.plain = &opt_force, },
+      .description = "Do not prompt for a confirmation to remove a key. "
+                     "Use with care, all keys matching the filter will be "
+                     "removed!", },
+    { .short_opt = 0, .long_opt = NULL, },
+};
+
+static const struct p11sak_arg p11sak_remove_key_args[] = {
+    { .name = "KEYTYPE", .type = ARG_TYPE_ENUM, .required = false,
+      .enum_values = p11sak_list_remove_key_keytypes,
+      .value.enum_value = &opt_keytype,
+      .description = "The type of the keys to select for removal (optional). "
+                     "If no key type is specified, all key types are "
+                     "selected.", },
+    { .name = NULL },
+};
+
+
+static const struct p11sak_cmd p11sak_commands[] = {
+    { .cmd = "generate-key", .cmd_short1 = "gen-key", .cmd_short2 = "gen",
+      .func = p11sak_generate_key,
+      .opts = p11sak_generate_key_opts, .args = p11sak_generate_key_args,
+      .description = "Generate a key.", .help = print_generate_key_attr_help },
+    { .cmd = "list-key", .cmd_short1 = "ls-key", .cmd_short2 = "ls",
+      .func = p11sak_list_key,
+      .opts = p11sak_list_key_opts, .args = p11sak_list_key_args,
+      .description = "List keys in the repository.",
+      .help = print_list_key_attr_help, },
+    { .cmd = "remove-key", .cmd_short1 = "rm-key", .cmd_short2 = "rm",
```

Comment:
Why are we using "remove-key" as a cmd and not "delete-key"? Looks like this command name was chosen in the very first p11sak version, but we think "delete-key" would be more intuitive.

Reply:
Well, it was 'remove-key' before, so I can't change this now. This would cause incompatibilities.
I also remember a discussion of remove vs. delete, and it turned out that remove would be the preferred word. I can't remember the details though....